### PR TITLE
Handle cases where the layout root is None. Fixes #6375.

### LINKED ISSUE #6375: LayoutTask panic 'no layout root' with root element in no namespace
```html
<script>
    document.removeChild(document.documentElement);
    document.appendChild(document.createElementNS(null, "html"));
</script>
```
causes
```
thread 'LayoutTask PipelineId(0)' panicked at 'no layout root', /rust/src/libcore/option.rs:330
stack backtrace:
   1:     0x7f5d49d73df3 - sys::backtrace::write::h2057a56abbba7b15Fcs
   2:     0x7f5d49d77731 - panicking::on_panic::h2be90c511aa31517uXw
   3:     0x7f5d49d6342a - rt::unwind::begin_unwind_inner::hc37996f90d433aa1eDw
   4:     0x7f5d49d63f17 - rt::unwind::begin_unwind_fmt::hd5e425b3f24d01eakCw
   5:     0x7f5d49d77086 - rust_begin_unwind
   6:     0x7f5d49da9944 - panicking::panic_fmt::hccac5539fcc4fdaai5B
   7:     0x7f5d47e4b1f2 - option::Option<T>::expect::h15271990684258674854
                        at /rust/src/libcore/macros.rs:27
   8:     0x7f5d47e4b01a - layout_task::LayoutTask::get_layout_root::hcc6b5fccf940b328uim
                        at /servo/components/layout/layout_task.rs:742
   9:     0x7f5d47e30c0a - layout_task::LayoutTask::handle_reflow::h90eaea98409c322bUtm
                        at /servo/components/layout/layout_task.rs:986
  10:     0x7f5d47e2c2cc - layout_task::LayoutTask::handle_request_helper::closure.43729
                        at /servo/components/layout/layout_task.rs:537
  11:     0x7f5d47e2c0fa - time::profile::h12768379534071970430
                        at /servo/components/profile_traits/time.rs:84
  12:     0x7f5d47e22ad1 - layout_task::LayoutTask::handle_request_helper::h8f50bdc79d4b7179c1l
                        at /servo/components/layout/layout_task.rs:534
  13:     0x7f5d47e09e12 - layout_task::LayoutTask::handle_request::ha8f2fa23477f6578eWl
                        at /servo/components/layout/layout_task.rs:452
  14:     0x7f5d47dcc9ff - layout_task::LayoutTask::start::h49949a15d2d582c8BUl
                        at /servo/components/layout/layout_task.rs:365
  15:     0x7f5d47dc9e7f - layout_task::LayoutTask.LayoutTaskFactory::create::closure.42433
                        at /servo/components/layout/layout_task.rs:245
  16:     0x7f5d47dc9824 - task::spawn_named_with_send_on_failure::closure.42422
                        at /servo/components/util/task.rs:31
  17:     0x7f5d47dc9734 - boxed::F.FnBox<A>::call_box::h4847288650633992202
                        at /rust/src/liballoc/boxed.rs:371
  18:     0x7f5d47dc912b - boxed::Box<FnBox<A, Output $u3d$$u20$R$GT$$u2b$$u20$Send$u20$$u2b$$u20$$u27$a$GT$.FnOnce$LT$A$GT$::call_once::h1851282661044141385
                        at /rust/src/liballoc/boxed.rs:387
  19:     0x7f5d47dc8bfb - thread::Builder::spawn_inner::closure.42395
                        at /rust/src/libstd/thread/mod.rs:345
  20:     0x7f5d47dc8b7e - rt::unwind::try::try_fn::__rust_abi::h14430643841743177991
                        at /rust/src/libstd/rt/unwind/mod.rs:158
  21:     0x7f5d47dc8b19 - rt::unwind::try::try_fn::h14430643841743177991
  22:     0x7f5d49d7c6b8 - rust_try_inner
  23:     0x7f5d49d7c6a5 - rust_try
  24:     0x7f5d49d71427 - rt::unwind::try::inner_try::hae2a3324f06bac5c7yw
  25:     0x7f5d47dc8a7f - rt::unwind::try::h2889938580734060236
                        at /rust/src/libstd/rt/unwind/mod.rs:130
  26:     0x7f5d47dc88cb - thread::Builder::spawn_inner::closure.42348
                        at /rust/src/libstd/thread/mod.rs:345
  27:     0x7f5d47dc945d - boxed::F.FnBox<A>::call_box::h12300694845718538607
                        at /rust/src/liballoc/boxed.rs:371
  28:     0x7f5d49d76221 - sys::thread::Thread::new::thread_start::h0114bb3965a53399jJv
  29:     0x7f5d46eece99 - start_thread
  30:     0x7f5d466f72ec - <unknown>
```

### PR DESCRIPTION
**Pull request by glennw**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7557)

### CODE REVIEW

**Comment by glennw** [comment]
r? @pcwalton 

cc @Ms2ger and @jdm

**Comment by pcwalton** [comment]
Yeah, this has come up again and again and I agree that being defensive is the right approach here. I'll look in depth at this patch in an hour or so.

File: components/layout/block.rs [hunk 1]
```diff
@@ -2963,7 +2963,7 @@ impl ISizeAndMarginsComputer for InlineBlockReplaced {
             MaybeAuto::Auto => false,
         });
 
-        let (mut computed_inline_size,
+        let (computed_inline_size,
```
Comment (pcwalton) [inline components/layout/block.rs]: Just silencing warnings, presumably?

**Comment by pcwalton** [comment]
```

/XMLHttpRequest/timeout-cors-async.htm
--------------------------------------
PASS expected FAIL
```

**Comment by glennw** [comment]
@bors-servo retry
 * #5740

**Comment by jdm** [comment]
For future reference, #6312 is a much better tracking issue for that intermittent failure.

**Pull request merged**